modules/new_profile/php/new_profile.class.inc

```diff
@@ -97,211 +54,83 @@ class New_Profile extends \NDB_Form
+        $this->result['endYear']    = $config->getSetting('endYear');
+        $this->result['ageMax']     = $config->getSetting('ageMax');
+        $this->result['ageMin']     = $config->getSetting('ageMin');
+        $this->result['dobFormat']  = $config->getSetting('dobFormat');
+        $this->result['edc']        = $config->getSetting('useEDC');
+        $this->result['useProject'] = $config->getSetting('useProjects');
+        $this->result['sex']        = [
+                                       'male'   => "Male",
+                                       'female' => "Female",
+                                      ];
+        $this->result['pscidSet']   = "false";
+        //get sites for the select dropdown
         $user_list_of_sites = $user->getData('CenterIDs');
         $num_sites          = count($user_list_of_sites);
-
-        if ($num_sites >1) {
-            $psc_labelOptions = array(null => '');
+        if ($num_sites >=1) {
             foreach ($user_list_of_sites as $key => $siteID) {
                 $center = $DB->pselectRow(
                     "SELECT CenterID as ID, Name FROM psc WHERE CenterID =:cid",
                     array('cid' => $siteID)
                 );
                 $psc_labelOptions[$siteID] = $center['Name'];
             }
-            $this->addSelect('psc', 'Site', $psc_labelOptions);
-
-            // site rules
-            $this->addRule('psc', 'Site is required', 'required');
+        } else {
+            $psc_labelOptions = array(null => '');
         }
-
+         $this->result['site'] = $psc_labelOptions;
+         //get projects for the select dropdown
+        $projects = \Utility::getProjectList();
+        foreach ($projects as $key=>$value) {
+             $list_of_projects[$key] = $value;
+        }
+        $this->result['project'] = $list_of_projects;
+         //get setting through pscid
         $PSCIDsettings = $config->getSetting('PSCID');
         if ($PSCIDsettings['generation'] == 'user') {
-            $this->addBasicText('PSCID', 'PSCID');
+            $result['pscidSet'] = "true";
         }
-
-        $this->form->addFormRule(array(&$this, '_validate'));
+        return true;
+    }
+    /**
+    * Call toJSON to display a json file on the page.
+    *
+    * @return string a json
+    */
+    function display()
+    {
+        return $this->toJSON();
     }
     /**
-     * Validate function
+     * Converts the results of this menu filter to a JSON format to be retrieved
+     * with ?format=json
      *
-     * @param array $values the value
+     * @return string a json encoded string of the headers and data from this table
+     */
+    function toJSON()
+    {
+        return json_encode($this->result);
+    }
+    /**
+     * Include additional JS files:
+     *  1. editForm.js - reactified form to update media
      *
-     * @return string
+     * @return array of javascript to be inserted
      */
-    function _validate($values)
```

Comment:
has the back-end validation been preserved and moved elsewhere?

Reply:
Yes, validateMatchDate in the jsx file.

Reply:
it should still be kept on the php side though i think. @ridz1208 thoughts?